src/esm_runscripts/workflow.py

```diff
@@ -1,8 +1,588 @@
+        ----------
+            self: class Workflow
+            phase_type: str (default or user)
+            attrib: str
+
+        Returns
+        -------
+            phases_attribs : list
+        """
+        if phase_type == 'user':
+            phases_attribs = [getattr(phase, attrib) for phase in self.user_phases]
+        else:
+            phases_attribs = [getattr(phase, attrib) for phase in self.phases]
+
+        return phases_attribs
+
+    def init_default_workflow(self, config):
+        """
+        Add workflow for precompute, compute, and tidy phases
+        etc information already here!
+
+        Parameters
+        ----------
+            self : Workflow object
+            config : dict
+
+        Returns
+        -------
+            self : Workflow object
+        """
+
+        workflow_phases = self.phases
+
+        # Calculating the number of tasks for each component/model
+        # needed for phase compute
+        tasks = calc_number_of_tasks(config)
+        # Initiate/create default workflow phase objects
+        # and reset/append to Workflow.phases variable
+        self.phases = []
+        for ind, phase in enumerate(workflow_phases):
+            self.phases.append(WorkflowPhase(phase))
+
+        for ind, phase in enumerate(self.phases):
+            if ind < self.num_phases - 1:
+                # Set run_before attrib of all phases (except last on)
+                # to the next phase name
+                phase.run_before = self.phases[ind+1].name
+                phase.next_submit.append(self.phases[ind+1].name)
+                phase.run_after = self.phases[ind-1].name
+            else:
+                # Set run_after attrib of last phase to previous phase name
+                phase.run_before = self.phases[0].name
+                phase.next_submit.append(self.phases[0].name)
+                phase.run_after = self.phases[ind-1].name
+
+            phase.cluster = phase.name
+            if phase.name in self.phases_to_submit_to_batch_system:
+                phase.batch_or_shell = 'batch'
+                phase.submit_to_batch_system = True
+                phase.run_on_queue = config["computer"]["partitions"]["compute"]["name"]
+                phase.nproc = tasks
+
+        # Set default workflow values
+        self.set_workflow_attrib("first_task_in_queue", self.phases[0].name)
+        self.set_workflow_attrib("last_task_in_queue", self.phases[-1].name)
+        self.set_workflow_attrib("next_run_triggered_by", self.phases[-1].name)
+
+        return self
+
+    def set_workflow_attrib(self, attrib, value):
+        """
+        Sets a workflow attribute.
+
+        Parameters
+        ----------
+            attrib : str
+            value :
+
+        Returns
+        -------
+            None
+        """
+
+        if type(getattr(self, attrib)).__name__ == "list":
+            self.__dict__[attrib].append(value)
+        else:
+            self.__setattr__(attrib, value)
```

Comment:
It can all be defined in `configs/esm_software/esm_runscripts/defaults.yaml` phase by phase as a dict.

One of the things that could be a useful check here is to see if the default workflow provided in the default contains this variables. If not trigger an error.

```suggestion
```